File: src/Entity/Commentaire.php
<?php

namespace App\Entity;

use App\Repository\CommentaireRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CommentaireRepository::class)]
#[ORM\Table(name: 'commentaire')]
class Commentaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Assert\NotBlank(message: 'L\'identifiant de l\'avis est obligatoire.')]
    #[Assert\Positive(message: 'L\'identifiant de l\'avis doit être positif.')]
    private int $avisId = 0;

    #[ORM\Column(type: 'integer')]
    private int $userId = 0;

    #[ORM\Column(type: 'text')]
    #[Assert\NotBlank(message: 'Le commentaire est obligatoire.')]
    #[Assert\Length(
        min: 2,
        max: 1000,
        minMessage: 'Le commentaire doit contenir au moins {{ limit }} caractères.',
        maxMessage: 'Le commentaire ne peut pas dépasser {{ limit }} caractères.'
    )]
    private string $contenu = '';

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getAvisId(): int
    {
        return $this->avisId;
    }

    public function setAvisId(int $avisId): static
    {
        $this->avisId = $avisId;
        return $this;
    }

    public function getUserId(): int
    {
        return $this->userId;
    }

    public function setUserId(int $userId): static
    {
        $this->userId = $userId;
        return $this;
    }

    public function getContenu(): string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): static
    {
        $this->contenu = $contenu;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(?\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }

}

File: nexora_java/src/main/java/com/pi/entities/Commentaire.java
package com.pi.entities;

import java.time.LocalDateTime;

public class Commentaire {
    private int id;
    private int avisId;
    private int userId;
    private String contenu;
    private LocalDateTime createdAt;

    public Commentaire() {}

    public Commentaire(int avisId, int userId, String contenu) {
        this.avisId = avisId;
        this.userId = userId;
        this.contenu = contenu;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getAvisId() { return avisId; }
    public void setAvisId(int avisId) { this.avisId = avisId; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    @Override
    public String toString() {
        return "Commentaire{id=" + id + ", avisId=" + avisId + ", userId=" + userId + "}";
    }
}